ptable/ptable.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Partition table
+ *
+ * Copyright 2020 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <errno.h>
+
+#include "ptable.h"
+
+
+#define MIN_PARTITIONS_SECTORS_NB    3
+
+
+static int ptable_checkType(uint8_t type)
+{
+	int res = EOK;
+
+	switch (type) {
+		case ptable_raw:
+		case ptable_meterfs:
+			break;
+
+		default:
+			res = -EINVAL;
+			break;
+	}
+
+	return res;
+}
+
+
+int ptable_verifyPartition(uint8_t id, const ptable_partition_t *pHeaders, const memory_properties_t *mem)
```

Comment:
I'm not sure why this is separate from `read` right now (I know, it can be reused in `write`).

I believe we have to check the whole partition table (if even one partitions would be invalid we should not trust the other entries). So maybe provide `ptable_verifyPartitionTable` interface?